telepresence/container.py

```diff
@@ -78,6 +78,13 @@ def run_docker_command(
     remote_env["TELEPRESENCE_ROOT"] = mount_dir
     remote_env["TELEPRESENCE_METHOD"] = "container"  # mostly just for tests :(
 
+    # Extract --publish flags and add them to the sshuttle container, which is
+    # responsible for defining the network entirely.
+    parser = argparse.ArgumentParser(allow_abbrev=False)
+    parser.add_argument("--publish", "-p", action="append", default=[])
+    publish_ns, docker_args = parser.parse_known_args(args.docker_run)
+    publish_args = ["-p={}".format(pub) for pub in publish_ns.publish]
```

Comment:
`run_docker_command` is probably already too long.  Maybe this new logic could go into a separate  function that returns `publish_args` and `docker_args`?  Not the nicest signature but it would keep the size of `run_docker_command` down and make it clear that `publish_ns` isn't used anywhere except in this logic.